fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/data/ApiGlobalErrorResponse.java

```diff
@@ -194,8 +194,7 @@ public ApiGlobalErrorResponse(final List<ApiParameterError> errors) {
         this.errors = errors;
     }
 
-    @XmlElementWrapper(name = "errors")
-    @XmlElement(name = "errorResponse")
+    @JsonProperty("errors")
```

Comment:
@ptuomola isn't this changing the structure of the error response? As I see it from the changes here:

Before:

```
<errors>
  <errorResponse>
    ...
  </errorResponse>
  <errorResponse>
    ...
  </errorResponse>
</errors>
```

After:

```
<errors>
  <apiParameterError>
    ...
  </apiParameterError>
  <apiParameterError>
    ...
  </apiParameterError>
</errors>
```
... if I'm missing something here and the errors still serialize to the same structure then please ignore.

Reply:
This code is actually used to serialise to JSON, not to XML. And with Jersey 2.x, the behaviour was changed: in Jersey 1.x, this would serialise to:

errors: [ "error1", "error2" ] 

ie take the name from the @XmlElementWrapper

In Jersey 2.x, the annotation @XmlElementWrapper is ignored, and instead the existing code serialises to:

errorResponse: [ "error1", "error2 ]

Which resulted in a lot of integration tests failing

So to restore the existing behaviour I changed the annotation to @JsonProperty (as this is anyway used for Json, not XML) - and for JSON there is no concept of a wrapper anyway. 

You can find discussion about this issue here:
https://github.com/FasterXML/jackson-modules-base/issues/45

Now if the same class is somewhere serialised to XML then I fully agree that this will introduce a regression. But I wasn't able to find any such cases - just the serialisation for JSON, for which this change actually retains the current behaviour...